library_generation/cli/entry_point.py

```diff
@@ -164,12 +185,17 @@ def __generate_repo_and_pr_description_impl(
         baseline_config=from_yaml(baseline_generation_config_path),
         current_config=from_yaml(current_generation_config_path),
     )
-    # pass None if we want to fully generate the repository.
+    # Pass None if we want to fully generate the repository.
     target_library_names = (
         config_change.get_changed_libraries()
         if not _needs_full_repo_generation(config_change=config_change)
         else None
     )
+    # Generate the union of two library lists if neither of the library list
+    # indicates a full generation.
+    # The deduplication is done in `generate_from_yaml`.
+    if (target_library_names is not None) and (include_libraries is not None):
```

Comment:
I feel this making the logic more complicated, if `include_libraries` is provided, it should just override `target_library_names`. Especially we plan to remove  `target_library_names` in the future.

Reply:
Yes, I also found it was a bit of confusing in this scenario.

I changed this to `library_names` takes preference if specified.